### ISSUE TITLE
redudant codemirror dependency

### ISSUE THREAD

**Issue by ikappaki** [nextjournal/command-bar#1]
Hi,

the command-bar library is dependent on `codemirror` at this early stage, in what appears to be code to support the demo.

https://github.com/nextjournal/command-bar/blob/ff66b824099b301e8885d3113191f9938b9313b5/src/nextjournal/command_bar.cljs#L2-L9

This means that early adopters for the lib will bring in an additional unnecessary dependency in their app, likely to cause conflicts if they also have a local codemirror setup, other than the unused code they bring in.

A solution is to relocate the relevant code to the demo.

PR to follow.

**Comment by zampino** [comment]
> This means that early adopters for the lib will bring in an additional unnecessary dependency in their app, likely to cause conflicts if they also have a local codemirror setup, other than the unused code they bring in.

I'm not sure I follow. Since clerk render deps are in the main profile, if shadow-cljs [resolves dependencies correctly](https://github.com/nextjournal/clerk/blob/4180ed31c2864687a770f6d4f625303bd8e75437/render/deps.cljs#L1-L8) you will still have codemirror deps pulled in, as per your remark in #2

>  the lib does not require any additional npm deps other than the one already defined in clerks deps.cljs.

so in theory we could just get rid of the deps.cljs in this repo.

Also I'm not sure this library was planned to be used outside of a codemirror context, but I'd rather wait for @philippamarkovics to take a look at this

> if they also have a local codemirror setup,

are you referring to a setup which is not involving a shadow build, like es module imports? Furthermore I think shadow will need those codemirror deps [at build time for compiling clojure-mode](https://github.com/nextjournal/command-bar/blob/7dfea8391960fa6f8a853b2ed32586834affff09/deps.edn#L7-L12) which is also part of the main deps.

**Comment by ikappaki** [comment]
> > This means that early adopters for the lib will bring in an additional unnecessary dependency in their app, likely to cause conflicts if they also have a local codemirror setup, other than the unused code they bring in.
> 
> I'm not sure I follow. Since clerk render deps are in the main profile, if shadow-cljs [resolves dependencies correctly](https://github.com/nextjournal/clerk/blob/4180ed31c2864687a770f6d4f625303bd8e75437/render/deps.cljs#L1-L8) you will still have codemirror deps pulled in, as per your remark in #2

You are right, I had a problem with the `@codemirror/view` dependency injected in my `package.json` that prevented `nextjournal/lang-clojure` from working when defined as a codemirror extension. Initially, I thought the problem was limited to the `command-button`'s `deps.cljs`. However, you correctly noted that the same dependency is also referenced in the clerk's rendering component.

The problem was resolved by removing the injected `@codemirror/view`, `lezer/common`, and `lezer/highlight` dependencies from the test application's `package.json`. The rationale behind this resolution remains unclear to me, having been deduced through trial and error.
 
> > the lib does not require any additional npm deps other than the one already defined in clerks deps.cljs.
> 
> so in theory we could just get rid of the deps.cljs in this repo.

Yes, I believe so, though I've been cautious about implementing significant modifications.

> Also I'm not sure this library was planned to be used outside of a codemirror context, but I'd rather wait for @philippamarkovics to take a look at this
> 
> > if they also have a local codemirror setup,
> 
> are you referring to a setup which is not involving a shadow build, like es module imports? 

I initially thought the issue with initializing a local codemirror instance in the test application stemmed from its use within the `command-button`, but this turned out to be incorrect as noted above.

Nevertheless, I believe the patch remains valid since it eliminates the unnecessary reference of an unused codemirror code.

Thanks
